PermissionVault/contracts/PermissionVault.sol

```diff
@@ -0,0 +1,348 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import {IERC721Receiver} from "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import {IERC1155} from "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import {IERC1155Receiver} from "@openzeppelin/contracts/token/ERC1155/IERC1155Receiver.sol";
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import {IPermissionVault} from "./interfaces/IPermissionVault.sol";
+
+error NotController();
+error NotEOA();
+error NotEnoughBalance();
+error ZeroAmount();
+error NotExistToken();
+error NotOwnerOf();
+
+contract PermissionVault is
+  IPermissionVault,
+  IERC721Receiver,
+  IERC1155Receiver,
+  Ownable,
+  AccessControl,
+  ReentrancyGuard
+{
+  using SafeERC20 for IERC20;
+
+  bytes32 public constant CONTROLLER_ROLE =
+    bytes32(keccak256("CONTROLLER_ROLE"));
+
+  /* -------------------------------------------------------------------------- */
+  /*                                   Events                                   */
+  /* -------------------------------------------------------------------------- */
+
+  /**
```

Comment:
Please explain about parameters in event
https://docs.soliditylang.org/en/v0.8.17/natspec-format.html